persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/LifecycleManager.java

```diff
@@ -0,0 +1,35 @@
+package org.infinispan.persistence.rocksdb;
+
+import org.infinispan.commons.configuration.attributes.Attribute;
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.StatisticsConfiguration;
+import org.infinispan.configuration.cache.StoreConfiguration;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.factories.annotations.InfinispanModule;
+import org.infinispan.lifecycle.ModuleLifecycle;
+import org.infinispan.persistence.rocksdb.configuration.RocksDBStoreConfiguration;
+import org.infinispan.persistence.rocksdb.logging.Log;
+import org.infinispan.persistence.rocksdb.metrics.StatisticsExporter;
+import org.infinispan.persistence.rocksdb.metrics.StatisticsExporterImpl;
+import org.infinispan.registry.InternalCacheRegistry;
+
+@InfinispanModule(name = "cachestore-rocksdb", requiredModules = "core")
+public class LifecycleManager implements ModuleLifecycle {
+    private static final Log log = LogFactory.getLog(LifecycleManager.class, Log.class);
+
+    @Override
+    public void cacheStarting(ComponentRegistry cr, Configuration configuration, String cacheName) {
```

Comment:
If I understand correctly, you have added `LifecycleManager` so that the `component-processor` will allow the metrics to work. However, I think this registration logic will be much clearner in the `RocksDbStore#init` method as you have direct access to the RockDb configuration. As `ModuleLifecyle` provides default implementations for all of it's methods, you can simple have `LifecycleManager` as a no-op implementation.

The RocksDbStore#init method would then look like:

```java
    @Override
    public void init(InitializationContext ctx) {
        this.configuration = ctx.getConfiguration();
        this.ctx = ctx;
        this.timeService = ctx.getTimeService();
        this.marshaller = ctx.getPersistenceMarshaller();
        this.semaphore = new Semaphore(Integer.MAX_VALUE, true);
        this.entryFactory = ctx.getMarshallableEntryFactory();
        
        Cache<K, V> cache = ctx.getCache();
        if (cache.getCacheConfiguration().statistics().enabled()) {
            GlobalComponentRegistry cr = cache.getCacheManager().getGlobalComponentRegistry();
            statisticsExporter = new StatisticsExporterImpl();
            cr.registerComponent(statisticsExporter, StatisticsExporter.class);
        }
    }
```

RocksDB is never used for internal caches, so you would not have to check this with the above approach.

Reply:
Actually, we can move this to the `#start` method. That way we can just call `new StatisticsExporterImpl(statistics)` and remove the need for the `StatisticsExporterImpl#init` method.

Reply:
I can't get the metrics to register if I do it at the init.

Reply:
@danberindei @anistor I am not familiar with stats registration, why can't this happen at store `init` or `start` time?

Reply:
It could be because the store `init` and `start` methods are called too late by `PersistenceManagerImpl`.

`CacheMetricsRegistration` registers all the components that it finds in the registry when it starts. If  `PersistenceManagerImpl` starts later, and only calls the store's `init` during `start`, then it would be too late.

My ideal solution would be to add a method in `CacheMetricsRegistration` that a component could use to expose itself as an MBean at any time, but for this PR perhaps it's best to keep the registration in `cacheStarting`.

 I don't remember if the MBean components